### DISCUSSION TITLE
Trigger playlist or media item via api

### DISCUSSION THREAD

**Discussion by t0mtaylor** [AzuraCast/AzuraCast#6495] [Q&A]
Hi

How do you Trigger a playlist or media item via api?

For example, if i wanted to force a playlist to play i could request the API, and based on the playlist rules it would then play

Thanks

**Comment by BusterNeece** [comment]
You can send a `PUT` request to `/api/station/(id)/files/batch` with the JSON body:

```json
{
  "do": "immediate",
  "files": [
    "path/to/file.mp3"
  ]
}
```

Replace `immediate` with `queue` to simply add the track to the queue, rather than playing it immediately.
